src/main/java/edu/cmu/cs/stage3/alice/core/geometry/Text3D.java

```diff
@@ -145,6 +149,19 @@ protected void updateGeometry() {
         }
     }
 
+    private GlyphVector getGlyphVector(String string) {
+        if (Bidi.requiresBidi(string.toCharArray(), 0, string.length())
+            && new Bidi(string, Bidi.DIRECTION_DEFAULT_LEFT_TO_RIGHT).isRightToLeft()) {
+            return m_font.layoutGlyphVector(
+                new FontRenderContext(new AffineTransform(), false, true),
+                string.toCharArray(),
+                0, string.length(), LAYOUT_RIGHT_TO_LEFT
+            );
+        } else {
+            return m_font.createGlyphVector(new FontRenderContext(new AffineTransform(),false,true), string);
```

Comment:
it isn't your fault since this is the OG code, but maybe add spaces between args?